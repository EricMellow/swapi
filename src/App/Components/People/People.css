.container {
  width: 80%;
  margin: 30px auto 10px auto;
  display: flex;
  justify-content: space-around;
  height: 65%;
  border: 2px solid white;
  box-shadow: 0px 0px 75px white;
  border-radius: 5px;
  flex-wrap: wrap;
  overflow-y: scroll;
}

.wait {
  font-family: 'Orbitron', sans-serif;
  font-size: 45px;
  text-align: center;
  margin-top: 30px;
  width: 80%;
}

.card {
  width: 220px;
  height: 275px;
  overflow-y: scroll;
  border: 2px solid #882233;
  text-align: center;
  margin: 10px;
  border-radius: 5px;
  background-color: rgb(255, 255, 255, .3);
}

p {
  font-size: 18px;
  margin: 15px 5px;
}

h3 {
  font-size: 22px;
  margin: 20px 5px;
  width: 65%;
  display: inline-block;
}

.fav {
  background-image: url(../button.png);
  background-repeat: no-repeat;
  background-size: cover;
  height: 50px;
  width: 50px;
  padding: 0;
  opacity: .5;
  margin: 0;
  margin-top: 25px;
}