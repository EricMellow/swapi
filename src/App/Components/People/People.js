import React from 'react';
import './People.css';
import PropTypes from 'prop-types';

const People = ({ cardData, toggleFavorite }) => {
  let peopleCards = [];
  if (cardData){
    peopleCards = cardData.map((person, index) => {
      return (
        <div key={person.name + index} className="card">
          <h3>{person.name}</h3>
          <button 
            className="fav" 
            onClick={() => toggleFavorite(person)}>
          </button>
          <p>Species: {person.species}</p>
          <p>Homeworld: {person.homeworld}</p>
          <p>Homeworld Population: {person.population}</p>
        </div>
      );
    });
  }

  return (
    <div className="container">
      {peopleCards.length === 0 && <div 
        className="wait">
      Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
      </div>}
      {peopleCards.length > 1 && peopleCards}
    </div>
  );
};


People.propTypes = {
  cardData: PropTypes.array,
  toggleFavorite: PropTypes.func
};

export default People;