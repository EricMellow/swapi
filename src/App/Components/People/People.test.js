import React from 'react';
import People from './People';
import { mockCleanedPeople } from '../../mockPeopleData';
import { shallow } from 'enzyme';

describe('People', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<People />);
    
  });

  it('should match the snapshot with no data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with correct card data', () => {
    const mockPeopleData = mockCleanedPeople;
   
    wrapper = shallow(<People cardData={mockPeopleData} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with error data', () => {
    const mockErrorData = [{
      name: "Oh no!",
      homeworld: "Something went wrong",
      species: ":(",
      population: "No data"
    }];

    wrapper = shallow(<People cardData={mockErrorData} />);

    expect(wrapper).toMatchSnapshot();
  });

  // it('should call toggleFavorite with the correct parameters when the button is clicked', () => {
  //   const mockData = [{
  //     id: "LukeSkywalker1",
  //     name: "Luke Skywalker",
  //     homeworld: "Test",
  //     species: "Human",
  //     population: 300000,
  //     type: 'person'
  //   }];
  //   const mockToggle = jest.fn();
    
  //   wrapper = shallow(<People cardData={mockData} toggleFavorite={mockToggle}/>);
    
  //   wrapper.instance().toggleFavorite()

  //   expect(toggleFavorite).toHaveBeenCalledWith(mockData);

  // });
});