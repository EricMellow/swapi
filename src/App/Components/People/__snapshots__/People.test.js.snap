// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match the snapshot with correct card data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    cardData={
        Array [
            Object {
              "homeworld": "Test",
              "name": "Luke Skywalker",
              "population": 300000,
              "species": "Human",
            },
            Object {
              "homeworld": "Test",
              "name": "Brett Bretterson",
              "population": 300000,
              "species": "Human",
            },
            Object {
              "homeworld": "Test",
              "name": "Mark Hammersmith",
              "population": 300000,
              "species": "Human",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <div
            className="card"
>
            <h3>
                        Luke Skywalker
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Species: 
                        Human
            </p>
            <p>
                        Homeworld: 
                        Test
            </p>
            <p>
                        Homeworld Population: 
                        300000
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        Brett Bretterson
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Species: 
                        Human
            </p>
            <p>
                        Homeworld: 
                        Test
            </p>
            <p>
                        Homeworld Population: 
                        300000
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        Mark Hammersmith
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Species: 
                        Human
            </p>
            <p>
                        Homeworld: 
                        Test
            </p>
            <p>
                        Homeworld Population: 
                        300000
            </p>
</div>,
        ],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "Luke Skywalker0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Luke Skywalker
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Species: 
              Human
</p>,
            <p>
              Homeworld: 
              Test
</p>,
            <p>
              Homeworld Population: 
              300000
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species: ",
                "Human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species: ",
              "Human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld: ",
                "Test",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld: ",
              "Test",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld Population: ",
                300000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld Population: ",
              300000,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Brett Bretterson1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Brett Bretterson
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Species: 
              Human
</p>,
            <p>
              Homeworld: 
              Test
</p>,
            <p>
              Homeworld Population: 
              300000
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brett Bretterson",
            },
            "ref": null,
            "rendered": "Brett Bretterson",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species: ",
                "Human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species: ",
              "Human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld: ",
                "Test",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld: ",
              "Test",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld Population: ",
                300000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld Population: ",
              300000,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Mark Hammersmith2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Mark Hammersmith
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Species: 
              Human
</p>,
            <p>
              Homeworld: 
              Test
</p>,
            <p>
              Homeworld Population: 
              300000
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mark Hammersmith",
            },
            "ref": null,
            "rendered": "Mark Hammersmith",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species: ",
                "Human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species: ",
              "Human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld: ",
                "Test",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld: ",
              "Test",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld Population: ",
                300000,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld Population: ",
              300000,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <div
              className="card"
>
              <h3>
                            Luke Skywalker
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Species: 
                            Human
              </p>
              <p>
                            Homeworld: 
                            Test
              </p>
              <p>
                            Homeworld Population: 
                            300000
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            Brett Bretterson
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Species: 
                            Human
              </p>
              <p>
                            Homeworld: 
                            Test
              </p>
              <p>
                            Homeworld Population: 
                            300000
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            Mark Hammersmith
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Species: 
                            Human
              </p>
              <p>
                            Homeworld: 
                            Test
              </p>
              <p>
                            Homeworld Population: 
                            300000
              </p>
</div>,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "Luke Skywalker0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Luke Skywalker
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Species: 
                Human
</p>,
              <p>
                Homeworld: 
                Test
</p>,
              <p>
                Homeworld Population: 
                300000
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species: ",
                  "Human",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species: ",
                "Human",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld: ",
                  "Test",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld: ",
                "Test",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld Population: ",
                  300000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld Population: ",
                300000,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Brett Bretterson1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Brett Bretterson
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Species: 
                Human
</p>,
              <p>
                Homeworld: 
                Test
</p>,
              <p>
                Homeworld Population: 
                300000
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brett Bretterson",
              },
              "ref": null,
              "rendered": "Brett Bretterson",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species: ",
                  "Human",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species: ",
                "Human",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld: ",
                  "Test",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld: ",
                "Test",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld Population: ",
                  300000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld Population: ",
                300000,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Mark Hammersmith2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Mark Hammersmith
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Species: 
                Human
</p>,
              <p>
                Homeworld: 
                Test
</p>,
              <p>
                Homeworld Population: 
                300000
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mark Hammersmith",
              },
              "ref": null,
              "rendered": "Mark Hammersmith",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Species: ",
                  "Human",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Species: ",
                "Human",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld: ",
                  "Test",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld: ",
                "Test",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld Population: ",
                  300000,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Homeworld Population: ",
                300000,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with error data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    cardData={
        Array [
            Object {
              "homeworld": "Something went wrong",
              "name": "Oh no!",
              "population": "No data",
              "species": ":(",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
